src/main/java/bio/terra/service/tabulardata/google/BigQueryPdao.java

```diff
@@ -1047,4 +1053,53 @@ private LegacySQLTypeName translateType(String datatype) {
             default: throw new IllegalArgumentException("Unknown datatype '" + datatype + "'");
         }
     }
+
+    private String externalTableName(String tableName, String suffix) {
```

Comment:
We've mostly banished the magic string prefixes from the PDAO, though I see the need for something like this one. Instead of having every method below take both a `tableName` and a `suffix`, immediately passing both into this method, could we push the computation into a setup step in the flight? Then the flight context can track the full external table name instead of the suffix.

Reply:
See my comment above. I prefer to have the flights only care about the suffix, and not have to remember things that are more the responsibility of the PDAO, like how to use the right prefix.

Reply:
@danxmoran Not sure what you mean "banished". We need to use name prefixes to avoid collisions with the things we let the users name.

@thathert I don't mind this as it is. If you _really_ want to encapsulate the name, you could make a step that calls PDAO to generate external table names. They could just be in the form `<pdao-prefix> + <clean UUID>`.